### PR TITLE
Build PDFs only on deploy

### PR DESCRIPTION
**Pull request by lzap**
I tried but this is the best I am able to achieve with GHA.

This makes PDFs to be built only when we deploy (merge into master). It is slow and we have never had a build failure for PDF. It is safe to do.

### CODE REVIEW

**Comment by lzap** [comment]
Do not merge, fighting with GHA syntax. YAML is good, BUT FOR DATA NOT CODE! :-)

https://github.community/t/accessing-environment-variable/211159

**Comment by ekohl** [comment]
> Do not merge, fighting with GHA syntax. YAML is good, BUT FOR DATA NOT CODE! :-)

I'm not sure I agree with "YAML is good" but I certainly agree that it's not good for code.

Overall :+1: for not building PDFs. I'm not sure anyone ever uses them.

**Comment by lzap** [comment]
> I'm not sure anyone ever uses them.

Yeah, the plan is to make sure they are working until we figure out how to publish them in the future. We can do it after merge, it is not critical now if they break.

**Comment by lzap** [comment]
Done, I could not figure it so I copy pasted the block twice. I at least broke it into multiple lines and shortened it a bit.

Also another change is that now we run HTML build and HTML local build+check in parallel. This speeds things even up. Total time is down from approx. 6 minutes to 2 minutes and that counts! This is now ready.

File: .github/workflows/deploy.yml
Comment (lzap) [inline .github/workflows/deploy.yml:15]: I also extracted these into variables and bumped Ruby to 2.7 version.

**Review by maximiliankolb — APPROVED** [review]
Fine for me. I would even +1 dropping the pdf document altogether.

**Comment by lzap** [comment]
For the record here is the deploy run that is supposed to build PDF actually: https://github.com/theforeman/foreman-documentation/actions/runs/1444557713

**Pull request merged**
